### ISSUE TITLE
Can't start apache2 after upgrading chroot from jessie to stretch

### ISSUE THREAD

**Issue by shelleyvadams** [dnschneid/crouton#3318]
#### Output of `sudo edit-chroot -all`
```
name: gdesk
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/gdesk...
crouton: version 1-20170619181610~master:980f090f
release: stretch
architecture: amd64
xmethod: xorg
targets: gnome
host: version 9592.37.0 (Official Build) beta-channel gnawty 
kernel: Linux localhost 4.4.64-09086-gaace1482168b #1 SMP PREEMPT Tue Jun 27 23:25:28 PDT 2017 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/gdesk...
```

#### Please describe your issue:
Last week, I upgraded my chroot from Debian 8 (jessie) to Debian 9 (stretch). The upgrade went smoothly, but I'm now unable to start Apache.

When I include the line `/etc/init.d/apache2 start` in `/etc/rc.local`, as described in the [LAMP section][lamp] of the [Running servers in crouton][serv] wiki page, I get the following when I enter the chroot:

```
Still running chroot's rc.local...
WARNING: /etc/rc.local failed with code 1
```

When I attempt to start the server with systemctl (i.e., `sudo systemctl start apache2.service`), that fails with:

```
Running in chroot, ignoring request.
```
 
Is there a way to ignore running in a chroot? Or a workaround that doesn't involve systemd? (I assume that's the source of the issue since the version [in stretch][sysd9] is considerably newer than [in jessie][sysd8].)

[lamp]: https://github.com/dnschneid/crouton/wiki/Running-servers-in-crouton#lamp
[serv]: https://github.com/dnschneid/crouton/wiki/Running-servers-in-crouton
[sysd8]: https://packages.debian.org/jessie/systemd
[sysd9]: https://packages.debian.org/stretch/systemd

**Comment by darethehair** [comment]
Interesting timing, since today I started working on an install of crouton onto a used Dell 11 Chromebook, and 'naturally' decided to try the newest version of Debian = 'stretch'.  I immediately ran into similar problems as described above.  I also had trouble installing 'php5'.  In desperation, I also installed 'jessie', and that worked fine.  My guess is that there is some sort of temporary glitch/oversight that is causing these problems.

Strangely, I also get the 'Running in chroot, ignoring request' when trying to start Apache2 with the systemctl command, even in Jessie, but it works fine (ultimately) using '/etc/init.d/apache2 start' (which I am used to).

On this topic, though, I was not able to figure out how to start apache2 (and others services) using the various methods -- and was forced to simply add entries to '/etc/rc.local'.  My guess is that there is some limitation of crouton that makes the regular methods not work (?).

**Comment by shelleyvadams** [comment]
I hadn't thought to try `/etc/init.d/apache2 start` directly, but it doesn't appear to work in Debian 9. The output:

```
[....] Starting Apache httpd web server: apache2Invoking 'systemctl start apache2'.
Use 'systemctl status apache2' for more info.
Running in chroot, ignoring request.
 failed!
[warn] The apache2 instance did not start within 20 seconds. Please read the log files to discover problems ... (warning).
```

`grep apache /var/log/messages` (outside chroot) yielded nothing.

**Comment by darethehair** [comment]
Yep, I was very puzzled and a bit disappointed, since I wanted to run the latest-and-greatest Debian 9 -- but I decided to try Debian 8, for which I had a bit more optimism.  I hope that Debian 9 is fixed so I can proceed with my original plan :)

**Comment by aleifuu** [comment]
I'm not sure. Do we have a solution for this yet ? ( Other than obvious downgrading to jessie ? )

**Comment by patelbhavesh** [comment]
I found the solution to this.
Run this command to start apache.Works fine in crouton.
apachectl -f /etc/apache2/apache2.conf

**Comment by ttresslar** [comment]
Hey, the above solution (semi) worked but I got this error message:
`AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using ::1. Set the 'ServerName' directive globally to suppress this message`

I did : `echo "ServerName localhost" | sudo tee /etc/apache2/conf-available/fqdn.conf && sudo a2enconf fqdn` as recommended by the [Troubleshooting Apache](https://help.ubuntu.com/community/ApacheMySQLPHP#Troubleshooting_Apache) website. Now works well!
